File: src/main/java/com/github/jaksonlin/testcraft/context/UnittestCase.java
package com.github.jaksonlin.testcraft.context;

import java.util.Collections;
import java.util.List;
import java.util.Map;

public class UnittestCase {
    private final Map<String, Object> values;

    public UnittestCase(Map<String, Object> values) {
        this.values = values;
    }

    public String getString(String key) {
        Object value = values.get(key);
        return value instanceof String ? (String) value : "";
    }

    public List<String> getStringList(String key) {
        Object value = values.get(key);
        if (value instanceof List<?>) {
            List<?> list = (List<?>) value;
            if (list.isEmpty() || list.get(0) instanceof String) {
                return (List<String>) value;
            }
        }
        return Collections.emptyList();
    }

    public String getStatus(String key) {
        Object value = values.get(key);
        return value instanceof String ? (String) value : "TODO";
    }

    public Map<String, Object> getValues() {
        return values;
    }
}

File: src/main/kotlin/com/github/jaksonlin/pitestintellij/context/UnittestCase.kt
package com.github.jaksonlin.pitestintellij.context

data class UnittestCase(
    val values: Map<String, Any?>
) {
    fun getString(key: String): String = values[key] as? String ?: ""
    fun getStringList(key: String): List<String> = values[key] as? List<String> ?: emptyList()
    fun getStatus(key: String): String = values[key] as? String ?: "TODO"
}